test/integration_test.py

```diff
@@ -1378,14 +1378,32 @@ def source_id_from_hit(hit: JSON) -> str:
         bundle_type = self._bundle_type(catalog)
         project_type = self._project_type(catalog)
 
-        hits = self._get_entities(catalog, project_type)
-        sources_found = set()
-        for hit in hits:
-            source_id = source_id_from_hit(hit)
-            sources_found.add(source_id)
-            self.assertEqual(source_id not in managed_access_source_ids,
-                             one(hit[project_type])['accessible'])
-        self.assertIsSubset(managed_access_source_ids, sources_found)
+        for accessible in True, False, None:
+            with self.subTest(accessible=accessible):
+                filters = None if accessible is None else {
+                    special_fields.accessible: {'is': [accessible]}
+                }
+                hits = self._get_entities(catalog, project_type, filters=filters)
+                accessible_sources = set()
+                inaccessible_sources = set()
+                for hit in hits:
+                    source_id = source_id_from_hit(hit)
+                    source_accessible = source_id not in managed_access_source_ids
+                    hit_accessible = one(hit[project_type])[special_fields.accessible]
+                    self.assertEqual(source_accessible, hit_accessible, hit['entryId'])
+                    if accessible is not None:
+                        self.assertEqual(accessible, hit_accessible)
+                    if source_accessible:
+                        accessible_sources.add(source_id)
+                    else:
+                        inaccessible_sources.add(source_id)
+                self.assertIsDisjoint(accessible_sources, inaccessible_sources)
+                self.assertIsDisjoint(managed_access_source_ids, accessible_sources)
+                self.assertEqual(set() if accessible else managed_access_source_ids,
+                                 inaccessible_sources)
+        # Verify that we reached the `None` iteration, which determines the
+        # return value
+        self.assertTrue(len(accessible_sources) > 0 and len(inaccessible_sources) > 0)
```

Comment:
I'd prefer my version if it is correct, simply because it is more direct.

```diff
Index: test/integration_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/integration_test.py b/test/integration_test.py
--- a/test/integration_test.py	(revision 22efbf8e4bc512b6a5b61470c85a3332a20dc07f)
+++ b/test/integration_test.py	(date 1713836659272)
@@ -1364,9 +1364,10 @@
                                      managed_access_source_ids: Set[str]
                                      ) -> JSONs:
         """
-        Test the managed access controls for the /index/bundles and
+        Test the managed-access controls for the /index/bundles and
         /index/projects endpoints
-        :return: hits for the managed access bundles
+
+        :return: hits for the managed-access bundles
         """
 
         special_fields = self.metadata_plugin(catalog).special_fields
@@ -1378,14 +1379,16 @@
         bundle_type = self._bundle_type(catalog)
         project_type = self._project_type(catalog)
 
-        for accessible in True, False, None:
+        unfiltered_hits = None
+        for accessible in None, False, True:
             with self.subTest(accessible=accessible):
                 filters = None if accessible is None else {
                     special_fields.accessible: {'is': [accessible]}
                 }
                 hits = self._get_entities(catalog, project_type, filters=filters)
-                accessible_sources = set()
-                inaccessible_sources = set()
+                if accessible is None:
+                    unfiltered_hits = hits
+                accessible_sources, inaccessible_sources = set(), set()
                 for hit in hits:
                     source_id = source_id_from_hit(hit)
                     source_accessible = source_id not in managed_access_source_ids
@@ -1401,9 +1404,7 @@
                 self.assertIsDisjoint(managed_access_source_ids, accessible_sources)
                 self.assertEqual(set() if accessible else managed_access_source_ids,
                                  inaccessible_sources)
-        # Verify that we reached the `None` iteration, which determines the
-        # return value
-        self.assertTrue(len(accessible_sources) > 0 and len(inaccessible_sources) > 0)
+        self.assertIsNotNone(unfiltered_hits, 'Cannot recover from subtest failure')
 
         bundle_fqids = self._get_indexed_bundles(catalog)
         hit_source_ids = {fqid.source.id for fqid in bundle_fqids}
@@ -1426,7 +1427,8 @@
             bundle_fqids = self._get_indexed_bundles(catalog, filters=source_filter)
         hit_source_ids = {fqid.source.id for fqid in bundle_fqids}
         self.assertEqual(managed_access_source_ids, hit_source_ids)
-        return hits
+
+        return unfiltered_hits
 
     def _test_managed_access_repository_files(self,
                                               catalog: CatalogName,
```